cypress/integration/frontend-testing.js

```diff
@@ -1,23 +1,54 @@
-function runFrontendTest(name) {
-    it(name, () => {
-        cy.window().then(win => {
-            return win.compilerExplorerFrontendTesting.run(name);
+import {it} from 'mocha';
+
+
+describe('UI testing', () => {
+    beforeEach(() => {
+        cy.visit('/');
+        cy.get('[data-cy="close-alert-btn"]:visible').click();
+        cy.get('[data-cy="new-compiler-dropdown-btn"]:visible').click();
+        // Shows every pane button even if the compiler does not support it
+        cy.get('[data-cy="new-pane-dropdown"]:visible button').each(($btn) => {
+            $btn.prop('disabled', false).show();
         });
     });
-}
 
-describe('Frontendtestresults', () => {
-    before(() => {
-        cy.visit('/');
+    afterEach(() => {
+        cy.window().then((win) => {
+            // Makes sure the pane did not output anything to stderr
+            expect(win.console.error).to.have.callCount(0);
+            expect(win.console.warn).to.have.callCount(0);
+        });
+        // Can't bring the site to a default state by clearing localStorage, so do it manually
+        cy.get('[data-cy="more-dropdown-btn"]').click();
+        cy.get('[data-cy="reset-ui-btn"]:visible').click();
     });
 
-    runFrontendTest('HelloWorld');
-});
+    function addPaneOpenTest(datacy, title) {
+        it(title + " pane", () => {
```

Comment:
Had to move it directly to this file. I couldn't get the ts file in `static/test/` to accept `it`